### PR TITLE
Add the ability to load ResultLists from files.

### PR DESCRIPTION
**Pull request by jeremykubica**
Add the ability to load the ResultLists from files. This will allow analysis functions, such as those in `create_stamps.py`, to load and work with a ResultList instead of tracking individual files. This also moves toward aggregating the result writing and loading into a single location for consistency.

### CODE REVIEW

File: src/kbmod/file_utils.py [hunk 1]
```diff
@@ -51,14 +51,20 @@ def load_csv_to_list(file_name, use_dtype=None):
...
         -------
         data : list of numpy arrays
             The data loaded.
         """
         if not Path(file_name).is_file():
-            raise ValueError(f"{file_name} does not exist")
+            if none_if_missing:
+                return None
+            else:
+                raise ValueError(f"{file_name} does not exist")
```
Comment (delucchi-cmu) [inline src/kbmod/file_utils.py:67]: Why not a FileNotFoundError?

File: src/kbmod/result_list.py [hunk 2]
```diff
@@ -456,3 +461,88 @@ def save_to_files(self, res_filepath, out_suffix):
...
+
+    Parameters
+    ----------
+    res_filepath : string
+        The directory in which to store the results.
+    suffix : string
+        The suffix to append to the output file name
+    all_mjd : list
+        A list of all the MJD timestamps (optional).
+        If not provided, the function looads from the all_times_ file.
```
Comment (delucchi-cmu) [inline src/kbmod/result_list.py:477]:
```suggestion
        If not provided, the function loads from the all_times_ file.
```

File: src/kbmod/result_list.py [hunk 3]
```diff
@@ -456,3 +461,88 @@ def save_to_files(self, res_filepath, out_suffix):
...
         np.save(ospath.join(res_filepath, f"all_ps_{out_suffix}.npy"), stamps_to_save)
+
+
+def load_result_list_from_files(res_filepath, suffix, all_mjd=None):
+    """Create a new ResultList from outputted files.
+
+    Parameters
+    ----------
+    res_filepath : string
+        The directory in which to store the results.
```
Comment (delucchi-cmu) [inline src/kbmod/result_list.py:472]:
```suggestion
        The directory in which the results are stored.
```

File: src/kbmod/result_list.py [hunk 4]
```diff
@@ -456,3 +461,88 @@ def save_to_files(self, res_filepath, out_suffix):
...
+    suffix : string
+        The suffix to append to the output file name
```
Comment (delucchi-cmu) [inline src/kbmod/result_list.py:474]:
```suggestion
        The suffix appended to the output file names
```

File: src/kbmod/result_list.py [hunk 5]
```diff
@@ -456,3 +461,88 @@ def save_to_files(self, res_filepath, out_suffix):
...
+    # Create the ResultList to store the data.
+    results = ResultList(all_times)
+
+    # Load the one required file (trajectories).
+    trajectories = FileUtils.load_results_file_as_trajectories(
+        ospath.join(res_filepath, f"results_{suffix}.txt")
+    )
+
+    # Treat the remaining files as optional. Note that the lightcurve (lc_) can be computed
+    # from psi + phi and times (time_) can be computed from lc_indices, so we do not need
```
Comment (delucchi-cmu) [inline src/kbmod/result_list.py:501]: If they don't need to be loaded, do they need to be written out in the first place?

File: tests/test_file_utils.py [hunk 6]
```diff
@@ -13,9 +13,14 @@ def test_save_load_csv(self):
             file_name = f"{dir_name}/data1.dat"
 
             # Check that there is nothing to load before saving the file.
+            # Be default FileUtils should raise a ValueError.
```
Comment (delucchi-cmu) [inline tests/test_file_utils.py:16]:
```suggestion
            # By default FileUtils should raise a ValueError.
```

File: tests/test_result_list.py [hunk 7]
```diff
@@ -312,6 +312,54 @@ def test_save_results(self):
...
+                row1_lc = row1.light_curve
+                row2_lc = row2.light_curve
+                for d in range(num_times):
+                    self.assertAlmostEqual(row1.psi_curve[d], row2.psi_curve[d])
+                    self.assertAlmostEqual(row1.phi_curve[d], row2.phi_curve[d])
+                    self.assertAlmostEqual(row1_lc[d], row2_lc[d])
+
+                # Check stamps.
+                self.assertEqual(row1.stamp is None, row2.stamp is None)
+                if row1.stamp is not None:
```
Comment (delucchi-cmu) [inline tests/test_result_list.py:354]: Looks like all the rows in the input/expected will have stamps. Can this conditional be removed?

File: src/kbmod/result_list.py (hunk 2 above)
Comment (jeremykubica) [reply src/kbmod/result_list.py:477]: Fixed

File: src/kbmod/result_list.py (hunk 3 above)
Comment (jeremykubica) [reply src/kbmod/result_list.py:472]: Fixed.

File: src/kbmod/result_list.py (hunk 5 above)
Comment (jeremykubica) [reply src/kbmod/result_list.py:501]: Currently they are used by other analysis scripts. We keep writing them for now so as to be backwards compatible.

File: tests/test_file_utils.py (hunk 6 above)
Comment (jeremykubica) [reply tests/test_file_utils.py:16]: Fixed. And made this FileNotFoundError.

File: tests/test_result_list.py (hunk 7 above)
Comment (jeremykubica) [reply tests/test_result_list.py:354]: Yes. Removed.

**Review by delucchi-cmu — APPROVED** [review]

**Pull request merged**